packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -560,8 +555,32 @@ class EditableText extends StatefulWidget {
        assert(rendererIgnoresPointer != null),
        assert(scrollPadding != null),
        assert(dragStartBehavior != null),
-       assert(toolbarOptions != null),
-       assert(clipBehavior != null),
+       toolbarOptions = toolbarOptions ?? (obscureText ?
+         (readOnly ?
+           // No point in even offering "Select All" in a read-only obscured
+           // field.
+           const ToolbarOptions() :
+           // Writable, but obscured
+           const ToolbarOptions(
+             selectAll: true,
+             paste: true,
+           )
+         ) :
+         (readOnly ?
+           // Read-only, not obscured
+           const ToolbarOptions(
+             selectAll: true,
+             copy: true,
+           ) :
+           // Writable, not obscured
+           const ToolbarOptions(
+             copy: true,
+             cut: true,
+             selectAll: true,
+             paste: true,
+           )
+         )
+       ),       assert(clipBehavior != null),
```

Comment:
Nit: Should this assert be on the next line?

Reply:
Yes.  Fixed.